python/tskit/metadata.py

```diff
@@ -61,8 +61,9 @@ def replace_root_refs(obj):
 META_SCHEMA["definitions"]["root"] = copy.deepcopy(META_SCHEMA)
 META_SCHEMA["codec"] = {"type": "string"}
 META_SCHEMA["required"] = ["codec"]
-# For interoperability reasons, force the top-level to be an object
-META_SCHEMA["properties"]["type"] = {"enum": ["object"]}
+# For interoperability reasons, force the top-level to be an object or union
+# of object and null
```

Comment:
It's not obvious why this isn't ``{"enum": ["object", "null"]}``, why not just a union of object and null?

Reply:
`{"enum": ["object", "null"]}` would allow `object` and `null` but wouldn't allow the union `["object", "null"]`

Reply:
I thought that's what we wanted? Metadata you decode is either an object or null?

Reply:
Yes, so we need to specify `["object", "null"]` as the `type` of the top-level. `{"enum": ["object", "null"]}` would not allow this as only `"object"` or `"null"` would be valid specifications.

Reply:
I'm being thick here, I know, I still don't get it. Probably easiest to talk me through it in person!

Reply:
This is the meta-schema, which defines what's allowed in the schema. The schema then defines what's allowed in the metadata. If you want either "object" or "null" allowed in the metadata, the schema says union of "object" and "null", and then the meta-schema must allow the schema to say unions are allowed for "object" and "null" (not just "object" or "null").

Reply:
(I recently read up on Guido's PEG parser stuff coming in python 3.9 (https://medium.com/@gvanrossum_83706/peg-parsing-series-de5d41b2ed60), so this thread caught my eye...)

Reply:
Riiiighhhtt. OK, I think I get it now. Thanks @grahamgower. In any case, if both of you are happy this is right then I'm happy too!

Reply:
Thanks for clearing that up @grahamgower! I admit the meta schema has had me head scratching quite a bit...